src/test/java/plugins/CompressArtifactsPluginTest.java

```diff
@@ -89,9 +91,25 @@ public void archiveLargerThan4GInTotal() throws Exception {
         FreeStyleJob job = jenkins.jobs.create();
         job.configure();
         job.addPublisher(ArtifactArchiver.class).includes("*");
-        job.addBuildStep(ShellBuildStep.class).command( // Generate archive larger than 4G
+        if (SystemUtils.IS_OS_UNIX) {
+            job.addBuildStep(ShellBuildStep.class).command( // Generate archive larger than 4G
                 "#!/bin/bash\nwget $JENKINS_URL/jnlpJars/jenkins-cli.jar -O stuff.jar; for i in {0..7000}; do cp -l stuff.jar stuff.${i}.jar; done"
-        );
+                                            );
+        }
+        else {
+            // On windows although we could create hard links this would exceed the number of hard links you can create for a given file. 
+            // and sparse files compress too well. 
+            //job.addBuildStep(BatchCommandBuildStep.class).command("for /l %%x in (0, 1, 7000) do fsutil file createnew stuff.%%x.jar 819200");
+            job.addBuildStep(BatchCommandBuildStep.class).command("powershell -command \"& { " +
+                                            "try { " +
```

Comment:
According to [this page](https://4sysops.com/archives/powershell-versions-and-their-windows-version/) the version of PowerShell on supported windows platforms (8+ or 2012+) is at least 3 and IIUC is installed by default - so the check would be redundant unless we are running on an unsupported platform at which point meh!